vllm/worker/worker.py

```diff
@@ -137,7 +138,8 @@ def init_device(self) -> None:
             _check_if_gpu_supports_dtype(self.model_config.dtype)
             gc.collect()
             torch.cuda.empty_cache()
-            self.init_gpu_memory = torch.cuda.mem_get_info()[0]
+            torch.cuda.reset_peak_memory_stats()
+            self.baseline_snapshot = MemorySnapshot()
```

Comment:
another key change: we also measure the non-torch memory before creating the vllm instance.

Reply:
Do we need to do this in `v1/worker/gpu_worker.py` as well?

Reply:
v1 does not use this `memory_profiling` utility yet. welcome to port it to v1 code path!